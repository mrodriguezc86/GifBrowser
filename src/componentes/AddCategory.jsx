import PropTypes from "prop-types";
import { useState } from "react";


export const AddCategory = ({onSubmitCategory}) => {
    const [inputValue, setInputValue] = useState('')

    const onInputChange = ({target}) => {
        setInputValue(target.value)
    }

    const onSubmit = (event) => {
        event.preventDefault();
        if(inputValue.trim().length<= 1) return;
        onSubmitCategory(inputValue)
        //setCategory(category => [inputValue, ...category])
        setInputValue('');
    }

    return(
        <form onSubmit={(event) => onSubmit(event)} aria-label="form">
            <input
                type='text'
                placeholder="Find gifs"
                value={inputValue}
                onChange={onInputChange}
            ></input>
        </form>
    )
}

AddCategory.propTypes = {
    onSubmitCategory: PropTypes.func.isRequired,
}