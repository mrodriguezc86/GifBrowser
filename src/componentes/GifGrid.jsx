import PropTypes from 'prop-types'
import { getGifs } from '../helpers/getGifs';
import { useEffect, useState } from 'react';
import { GifItem } from './GifItem';
import { useFetchGifs } from '../hooks/useFetchGifs';

export const GifGrid = ({category}) => {

  const {images, isLoading} = useFetchGifs( category )
   

  return (
    <div>
      <h3>{category}</h3>

      {isLoading && (<h2>Loading...</h2>)}

      <div className='card-grid'>
        {images.map((image)=>{
          return <GifItem key={image.id} {...image}></GifItem>
        })}

      </div>
      
    
    </div>
  )
}

GifGrid.propTypes = {
  category: PropTypes.string.isRequired,
}

