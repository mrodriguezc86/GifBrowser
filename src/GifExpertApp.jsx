import { useState } from "react"
import {AddCategory, GifGrid} from './componentes'

export const GifExpertApp = () => {

    const [category, setCategory] = useState([])

    const onSubmitCategory = (inputValue) => {
        if(category.includes(inputValue))return;
        setCategory([inputValue, ...category])
    }

  return (
    <>
        <h1>Gif Browser</h1>
        <AddCategory onSubmitCategory={onSubmitCategory}/>
        
       
            {
                category.map(category => (
                    <GifGrid key={category} category={category}></GifGrid>
                ))
            }
           
        
    </>
  )
}
