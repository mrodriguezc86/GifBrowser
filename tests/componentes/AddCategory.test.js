import { fireEvent, render, screen } from "@testing-library/react"
import { AddCategory } from "../../src/componentes"

describe('test for <AddCategory/>', () => { 
    test('should match snapshot', () => { 
        const {container} = render(<AddCategory onSubmitCategory={ ()=>{} } />)
        //screen.debug();
        expect(container).toMatchSnapshot();
     }) 

     test('should change the texbox value', () => { 
        render(<AddCategory onSubmitCategory={ ()=>{} } />)
        const input = screen.getByRole('textbox')

        fireEvent.input(input, { target: { value: "Hello" } })
        expect(input.value).toBe('Hello')
      })

      test('should call onSubmitCategory if the input has a value', () => { 
        const inputValue = 'Hello'
        const onSubmitCategory = jest.fn();

        render(<AddCategory onSubmitCategory={onSubmitCategory}/>);

        const input = screen.getByRole('textbox');
        const form = screen.getByRole('form');

        fireEvent.input( input, {target: {value: inputValue}})
        fireEvent.submit(form);
        //screen.debug();

        //clear the textbox
        expect(input.value).toBe('');

        //onSubmitCategory called
        expect(onSubmitCategory).toHaveBeenCalled()
        expect(onSubmitCategory).toHaveBeenCalledTimes(1)
        expect(onSubmitCategory).toHaveBeenCalledWith(inputValue)

       })

       test('should not call onSubmitCategory if input is empty', () => { 
          const inputValue = ''
          const onSubmitCategory = jest.fn()

          render(<AddCategory onSubmitCategory={onSubmitCategory}/>);

          const input = screen.getByRole('textbox');
          const form = screen.getByRole('form');
  
          fireEvent.input( input, {target: {value: inputValue}})
          fireEvent.submit(form);
          //screen.debug();

          //onSubmitCategory not called
          expect(onSubmitCategory).not.toHaveBeenCalled()
          expect(onSubmitCategory).toHaveBeenCalledTimes(0)

        })
 })  