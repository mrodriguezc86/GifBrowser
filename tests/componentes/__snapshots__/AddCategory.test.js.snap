// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test for <AddCategory/> should match snapshot 1`] = `
<div>
  <form
    aria-label="form"
  >
    <input
      placeholder="Find gifs"
      type="text"
      value=""
    />
  </form>
</div>
`;
